### PR TITLE
Documented essentially everything from init routine to end of nemesis decompressor using s2disasm

### PR DESCRIPTION
**Pull request by GabrielRavier**
There isn't much to say except that I changed the checksum to the REV00 value so that it's easier to check if the rom is bit perfect

### CODE REVIEW

**Comment by GabrielRavier** [comment]
I know fixheader fixes the checksum : That's the point. When I see `Reported Size:    6FEFF Reported Checksum: 264A
 Size applied:    7FFFF  Checksum Applied: 264A`
I know that I didn't accidentally change the code and thus changed the applied checksum accidentally.
But if I see `Reported Size:    6FEFF Reported Checksum: 264A
 Size applied:    7FFFF  Checksum Applied: 271F`
Then I know I borked up something. The change may not be of much usefulness to anyone that isn't just doing documentation, but it's not harming them or anything. I have no problem with changing it back if you still consider it's a problem though

**Comment by Clownacy** [comment]
I was waiting a few days to see if anyone else had any issues with the pull request before I merged it. Since flamewing spoke up about the checksum, I'm not sure what to do with it right now.

**Comment by flamewing** [comment]
If you want to go the route of hard-coding the checksum, you should make it depend on the selected revision (all 3 of them) instead of using rev00 checksum.

File: sonic.asm [hunk 1]
```diff
@@ -54,34 +102,38 @@ loc_E0:
...
 		dc.l $20202020		; SRAM start ($200001)
 		dc.l $20202020		; SRAM end ($20xxxx)
-Notes:		dc.b "                                                    "
-Region:		dc.b "JUE             " ; Region
+Notes:		dc.b "                                                    " ; Notes (unused, anything can be put in this space, but it has to be 52 bytes.)
+Region:		dc.b "JUE             " ; Region (Country code)
+EndOfHeader:
 
 ; ===========================================================================
+; Crash/Freeze the 68000. Note that the Z80 continues to run, so the music keeps playing.
```
Comment (FraGag) [inline sonic.asm]: Sonic 1 plays music on the 68000, so "the music keeps playing" is wrong.

Comment (GabrielRavier) [reply sonic.asm]: oops changing that now

File: sonic.asm [hunk 2]
```diff
@@ -93,69 +145,71 @@ EntryPoint:
...
-		beq.s	SkipSecurity
+		beq.s	SkipSecurity	; If the console has no TMSS, skip the security stuff.
 		move.l	#'SEGA',$2F00(a1) ; move "SEGA" to TMSS register ($A14000)
 
 SkipSecurity:
-		move.w	(a4),d0
-		moveq	#0,d0
-		movea.l	d0,a6
-		move.l	a6,usp		; set usp to 0
+		move.w	(a4),d0	; check if VDP works
```
[the remark this answers is not in the document]
Comment (GabrielRavier) [reply sonic.asm]: I'll correct that too (in both disasms)

File: sonic.asm [hunk 3]
```diff
@@ -1119,22 +1119,24 @@ TilemapToVRAM:
...
+
+; ---------------------------------------------------------------------------
+; Subroutine to load pattern load cues (aka to queue pattern load requests)
 ; ---------------------------------------------------------------------------
-; Subroutines to load pattern load cues
 
-; input:
-;	d0 = pattern load cue number
+; ARGUMENTS
+; d0 = index of PLC lis
```
Comment (flamewing) [inline sonic.asm]: typo: lis

Comment (GabrielRavier) [reply sonic.asm]: oops, gonna solve that as well as making the checksum comment clearer (as in add a note about it only being a signal for positives, not negatives)

**Comment by GabrielRavier** [comment]
So yeah i did that. And I corrected that comma. @flamewing ?

**Pull request merged**